appcues/src/main/java/com/appcues/ui/modal/SharedModal.kt

```diff
@@ -31,6 +40,42 @@ internal fun dialogExitTransition(): ExitTransition {
     return fadeOut(tween(durationMillis = 100))
 }
 
+internal fun slideOutEnterTransition(initial: TransitionEdge, horizontalPadding: Int, verticalPadding: Int): EnterTransition {
+    val slide = slideIn(
+        initialOffset = { initial.toOffset(it, horizontalPadding, verticalPadding) },
+        animationSpec = tween(durationMillis = 1000)
+    )
+
+    return if (initial == CENTER) {
+        slide + fadeIn(tween(1000))
+    } else {
+        slide
+    }
+}
+
+internal fun slideOutExitTransition(target: TransitionEdge, horizontalPadding: Int, verticalPadding: Int): ExitTransition {
+    val slide = slideOut(
+        targetOffset = { target.toOffset(it, horizontalPadding, verticalPadding) },
+        animationSpec = tween(durationMillis = 300)
+    )
+
+    return if (target == CENTER) {
```

Comment:
this is the special case "edge" `CENTER` - used when the slideout is positioned at center+center rather than truly on an edge. In this case, we do a fade in/out as well as a transition up by half the content height.